src/main/java/org/example/Extractor.java

```diff
@@ -1,26 +1,67 @@
 package org.example;
 
+import org.jsoup.Connection;
 import org.jsoup.Jsoup;
 import org.jsoup.nodes.Document;
 import org.jsoup.nodes.Element;
-
+import org.jsoup.select.Elements;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 
 public class Extractor {
-    public List<Chair> getChairs(String url){
+    public List<Chair> getChairs(String url) throws IOException {
+
 
         String role, name, email;
         List<Chair> chairs = new ArrayList<>();
-        //todo
+
+        Connection connection = Jsoup.connect("https://www.ualberta.ca/computing-science/faculty-and-staff/index.html");
+        Document document = connection.get();
+
+        Elements Chairs = document.getElementsByTag("tr");
+        for (Element Chair : Chairs) {
+
+            Elements td0 = Chair.getElementsByAttribute("Style");
+            Elements td1 = Chair.getElementsByTag("strong");
+            Elements td2 = Chair.getElementsByAttribute("href");
+            String Role = td0.text().trim();
+            if (Role.contains("Chair")){
+
+                String Name = td1.text().trim();
+                String Email = td2.text().trim();
+                Chair chair = new Chair(Name, Role, Email);
+
+                chairs.add(chair);
+            }
+        }
+
         return chairs;
     }
-    public List<ResearchLab> findResearchLabs(String url) {
+
+    public List<ResearchLab> findResearchLabs(String url) throws IOException {
 
         List<ResearchLab> researchLabs = new ArrayList<>();
-        //todo
+
+        Connection connection = Jsoup.connect("https://www.ualberta.ca/computing-science/research/index.html");
+        Document document = connection.get();
+
+        Elements Labs = document.getElementsByTag("a");
+
+
+        for (Element Lab : Labs) {
+            String LabName = Lab.text().trim();
+            String LabLink = Lab.attr("abs:href");
+
+            String Prefix = "https://www.ualberta.ca/computing-science/research/research-areas/";
+
+            if (LabLink.startsWith(Prefix)) {
```

Comment:
This will also print the lab name: "Research Areas" and lab link: "https://www.ualberta.ca/computing-science/research/research-areas/index.html" which is not our requirement. Can you put a logic so that it doesn't print them? For example, you can check if the link ends with "/index.html" and then you can ignore that.

Reply:
Added code to exclude reaserach areas